### PR TITLE
Maui eBook Communication Between Loosely Coupled Components

### PR DESCRIPTION
**Pull request by michaelstonis**
## Summary

Communication Between Loosely Coupled Components eBook chapter

### CODE REVIEW

File: docs/architecture/maui/communicating-between-components.md [hunk 1]
```diff
@@ -1,9 +1,112 @@
...
+
+The eShopOnContainers mobile app uses the `MessagingCenter` class to communicate between loosely coupled components. The app defines a single message. The `AddProduct` message is published by the `CatalogViewModel` class when an item is added to the shopping basket. In return, the `CatalogView` class subscribes to the message and uses this to highlight the product adds with an animation in response.
+
+> [!NOTE]
+> While the `MessagingCenter` class permits communication between loosely-coupled classes, it does not offer the only architectural solution to this issue. For example, communication between a view model and a view can also be achieved by the binding engine and through property change notifications. In addition, communication between two view models can also be achieved by passing data during navigation.
+
+In the eShopOnContainers mobile app, `MessagingCenter` is used to update in the UI in response to an action occurring in another class. Therefore, messages are published on the UI thread, with subscribers receiving the message on the same thread.
+
+> [!TIP]
+> Optional information to help a user be more successfulMarshal to the UI thread when performing UI updates
```
Comment (IEvangelist) [inline docs/architecture/maui/communicating-between-components.md:35]: I'm not sure what you're trying to say here, you need to revise this.

File: docs/architecture/maui/communicating-between-components.md [hunk 2]
```diff
@@ -1,9 +1,112 @@
...
+
+## Subscribing to a message
+
+Subscribers can register to receive a message using one of the `MessagingCenter.Subscribe` overloads. The following code example demonstrates how the eShopOnContainers mobile app subscribes to, and processes, the `AddProduct` message:
+
+```csharp
+MessagingCenter.Subscribe<CatalogViewModel>(
+    this,
+    MessengerKeys.AddProduct,
+    (sender) =>
```
Comment (IEvangelist) [inline docs/architecture/maui/communicating-between-components.md:83]:
```suggestion
    _ =>
```

If the `sender` isn't used, use a discard `_`.

File: docs/architecture/maui/communicating-between-components.md [hunk 3]
```diff
@@ -1,9 +1,112 @@
...
+        MainThread.BeginInvokeOnMainThread(
+            async () =>
+            {
+                await badge.ScaleTo(1.2);
+                await badge.ScaleTo(1.0);
+            });
+    });
+```
+
+In this example, the Subscribe method subscribes to the `AddProduct` message, and executes a callback delegate in response to receiving the message. This callback delegate, specified as a lambda expression, executes code that updates the UI.
```
Comment (IEvangelist) [inline docs/architecture/maui/communicating-between-components.md:94]:
```suggestion
In the preceding example, the Subscribe method subscribes to the `AddProduct` message, and executes a callback delegate in response to receiving the message. This callback delegate, specified as a lambda expression, executes code that updates the UI.
```

File: docs/architecture/maui/communicating-between-components.md [hunk 4]
```diff
@@ -1,9 +1,112 @@
...
+If payload data is supplied, don't attempt to modify the payload data from within a callback delegate because several threads could be accessing the received data simultaneously. In this scenario, the payload data should be immutable to avoid concurrency errors.
+
+A subscriber might not need to handle every instance of a published message, and this can be controlled by the generic type arguments that are specified on the Subscribe method. In this example, the subscriber will only receive `AddProduct` messages that are sent from the `CatalogViewModel` class.
+
+## Unsubscribing from a message
+
+Subscribers can unsubscribe from messages they no longer want to receive. This is achieved with one of the `MessagingCenter`.Unsubscribe overloads, as demonstrated in the following code example:
+
+```csharp
+MessagingCenter.Unsubscribe<CatalogViewModel>(this, MessengerKeys.AddProduct);
```
Comment (IEvangelist) [inline docs/architecture/maui/communicating-between-components.md:105]:
```suggestion
MessagingCenter.Unsubscribe<CatalogViewModel>(
    this, MessengerKeys.AddProduct);
```

**Review by IEvangelist — COMMENTED** [review]
A few nits.

File: docs/architecture/maui/communicating-between-components.md (hunk 1 above)
Comment (michaelstonis) [reply docs/architecture/maui/communicating-between-components.md:35]: I missed this. This was the VSCode suggestion that got brought in accidentally. I will update the language a bit more though.

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**